modules/ethereum-contract/builtin/src/lib.rs

```diff
@@ -45,35 +49,272 @@ pub struct Header {
 }
 
 /// GRANDPA validators set change signal.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub struct ValidatorsSetSignal {
 	/// Signal delay.
 	pub delay: BlockNumber,
 	/// New validators set.
 	pub validators: Vec<u8>,
 }
 
+/// Convert from U256 to BlockNumber. Fails if `U256` value isn't fitting within `BlockNumber`
+/// limits (the runtime referenced by this module uses u32 as `BlockNumber`).
```

Comment:
What would happen if we changed the runtime to use `u64` for `BlockNumber` without updating this function?

Reply:
Nothing good would happen. This code references specific runtime, so whenever you change the runtime, it must be updated